griptape/utils/prompt_stack.py

```diff
@@ -1,23 +1,24 @@
 from __future__ import annotations
-from typing import Optional, TYPE_CHECKING
-from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional
 from attr import define, field
 
+from griptape.mixins import SerializableMixin
+
 if TYPE_CHECKING:
     from griptape.memory.structure import ConversationMemory
 
 
 @define
-class PromptStack:
+class PromptStack(SerializableMixin):
     GENERIC_ROLE = "generic"
     USER_ROLE = "user"
     ASSISTANT_ROLE = "assistant"
     SYSTEM_ROLE = "system"
 
-    @dataclass
-    class Input:
-        content: str | list[dict]
-        role: str
+    @define
+    class Input(SerializableMixin):
+        content: str = field(metadata={"serializable": True})
+        role: str = field(metadata={"serializable": True})
```

Comment:
minor, but is there any value in defining a reusable import to denote serializable/non-serializable fields, something like `field(**utils.serializable)`? Reduces _some_ repetition and addresses the issue that a typo in "serializable" wouldn't be caught by a type check, but it prevents us from easily using `metadata` for other uses.

Reply:
Interesting point... Another option could be to create a custom field as outlined [here](https://www.attrs.org/en/stable/extending.html#metadata). Something like `serializable_field(...)`. Not sure I love this one though, defining a serializable metadata constant somewhere might be easier.